testing/scenario_app/lib/src/animated_color_square.dart

```diff
@@ -14,21 +14,18 @@ import 'scenario.dart';
 /// that are constantly changing.
 class AnimatedColorSquareScenario extends Scenario {
   /// Creates the AnimatedColorSquare scenario.
-  ///
-  /// The [dispatcher] parameter must not be null.
-  AnimatedColorSquareScenario(PlatformDispatcher dispatcher)
-      : super(dispatcher);
+  AnimatedColorSquareScenario(super.view);
 
   static const double _squareSize = 200;
   /// Used to animate the red value in the color of the square.
   final _NumberSwinger<int> _r = _NumberSwinger<int>(0, 255);
-  _NumberSwinger<double> _top = _NumberSwinger<double>(
+  late _NumberSwinger<double> _top = _NumberSwinger<double>(
```

Comment:
My knowledge of Dart is failing me. Is the `late` necessary here to initialize the field after the constructor?

Reply:
With `late` this gets initialized when the field is first accessed. That allows us to access other fields (here `view`) from the initializer. Without it, the field is initialized at construction time and accessing other non-static fields would not be allowed.